src/components/reusable/datePicker/datepicker.ts

```diff
@@ -1,73 +1,161 @@
-import { LitElement, html, PropertyValues } from 'lit';
+import { html, LitElement, PropertyValues } from 'lit';
 import { customElement, property, query, state } from 'lit/decorators.js';
-import { ifDefined } from 'lit/directives/if-defined.js';
 import { classMap } from 'lit/directives/class-map.js';
-import { DATE_PICKER_TYPES } from './defs';
 import { FormMixin } from '../../../common/mixins/form-input';
-import DatePickerScss from './datepicker.scss';
-import { deepmerge } from 'deepmerge-ts';
+import { unsafeSVG } from 'lit-html/directives/unsafe-svg.js';
+import {
+  injectFlatpickrStyles,
+  langsArray,
+  initializeSingleAnchorFlatpickr,
+  getFlatpickrOptions,
+  getPlaceholder,
+  preventFlatpickrOpen,
+  handleInputClick,
+  handleInputFocus,
+  updateEnableTime,
+  setCalendarAttributes,
+  loadLocale,
+  emitValue,
+} from '../../../common/helpers/flatpickr';
+
+import flatpickr from 'flatpickr';
+import { BaseOptions } from 'flatpickr/dist/types/options';
+
+import DatePickerStyles from './datepicker.scss';
+import ShidokaFlatpickrTheme from '../../../common/scss/shidoka-flatpickr-theme.scss';
+
+import '@kyndryl-design-system/shidoka-foundation/components/icon';
+import errorIcon from '@kyndryl-design-system/shidoka-icons/svg/monochrome/16/close-filled.svg';
+import calendarIcon from '@kyndryl-design-system/shidoka-icons/svg/monochrome/16/calendar.svg';
+
+type SupportedLocale = (typeof langsArray)[number];
 
 const _defaultTextStrings = {
   requiredText: 'Required',
 };
 
 /**
- * Datepicker.
- * @fires on-input - Captures the input event and emits the selected value and original event details.
- * @prop {string} minDate - Mimimum date in YYYY-MM-DD or YYYY-MM-DDThh:mm format. If the value isn't a possible date string in the format, then the element has no minimum date value.
- * @prop {string} maxDate - Maximum date in YYYY-MM-DD or YYYY-MM-DDThh:mm format. If the value isn't a possible date string in the format, then the element has no maximum date value
- * @slot unnamed - Slot for label text.
+ * Datepicker: uses flatpickr datetime picker library -- `https://flatpickr.js.org`
+ * @fires on-change - Captures the input event and emits the selected value and original event details.
+ * @slot tooltip - Slot for tooltip.
  */
-
 @customElement('kyn-date-picker')
 export class DatePicker extends FormMixin(LitElement) {
-  static override styles = DatePickerScss;
+  static override styles = [DatePickerStyles, ShidokaFlatpickrTheme];
 
-  /** Datepicker size. "sm", "md", or "lg". */
+  /** Sets datepicker attribute name (ex: `contact-form-date-picker`). */
   @property({ type: String })
-  size = 'md';
+  nameAttr = '';
```

Comment:
Should this be `name` to remain consistent with other form components?

Reply:
✅